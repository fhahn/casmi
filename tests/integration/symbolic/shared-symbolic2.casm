// symbolic
CASM test

init main

function (symbolic) a: -> Int // symbol
function (symbolic) b: -> Int // symbol

rule main = {|
  a := BVadd_result( 20, b, b )

  program( self ) := undef
|}
