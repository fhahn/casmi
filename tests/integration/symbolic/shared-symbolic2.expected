forklog:
tff(symbolNext, type, sym2: $int).
fof(id0,hypothesis,stb(1,sym2)).%CREATE: b
tff(symbolNext, type, sym4: $int).
fof(id1,hypothesis,sta(1,sym4)).%CREATE: a
tff(symbolNext, type, sym3: $int).
fof(id2,hypothesis,fadd_result(20, sym2, sym2, sym3)).
fof(id3,hypothesis,sta(2,sym3)).%UPDATE: a
fof(id4,hypothesis,stb(2,sym2)).%SYMBOLIC: b
fof(final0,hypothesis,sta(0,sym3)).%FINAL: a
fof(final1,hypothesis,stb(0,sym2)).%FINAL: b

